### PR TITLE
refactor fleet model Closes #48

### LINKED ISSUE #34: Add entity models for a single plane and for a flight
We previously had models for our fleet and flight schedule.

Add entity models for a single plane and for a flight, and make sure that we also have a model to retrieve shared data from the WACKY server.

The entity models will be similar to what we build for lab 7.

Business rules to enforce, for entities, are the same as for assignment 1, with the clarification that **recognized airplanes** are ones in the data from the WACKY server.

### LINKED ISSUE #48: Rework of fleet model
Refactor the fleet model (and possibly the plane model) according to this:

image

**Include unit testing as well**

### PR DESCRIPTION
**Pull request by kculala**
- refactor fleet model and add unit test. 
- wrapped responses with WackyModel functions in json_decode()

### CODE REVIEW

File: application/models/FleetModel.php [hunk 1]
```diff
@@ -19,24 +19,31 @@ public function all()
...
                 {
-                    foreach (get_object_vars($wackyPlane) as $prop => $val)
-                        if (!property_exists($plane, $prop))
-                            $plane->$prop = $val;
+                    $tempPlane = array();
+                    $tempPlane['id']           = $plane->id;
+                    $tempPlane['key']          = $plane->airplaneCode;
+                    $tempPlane['model']        = $wackyPlane->model;
+                    $tempPlane['manufacturer'] = $wackyPlane->manufacturer;
+                    array_push($fleet, $tempPlane);
```
Comment (achlin) [inline application/models/FleetModel.php]: Why was this needed? Didn't the code you removed do the same thing?

File: application/models/FleetModel.php [hunk 2]
```diff
@@ -45,12 +52,33 @@ public function all()
...
         $plane = parent::get($id, $key2);
-        $wackyPlane = json_decode(file_get_contents(WACKY_SERVER_URI_BASE . '/airplanes/' . $plane->airplaneCode));
+        $wackyPlane = $this->wackyModel->getAirplane($plane->airplaneCode);
+        $wackyPlane->id = $id;
+        return $wackyPlane;
+    }
 
-        foreach (get_object_vars($wackyPlane) as $prop => $val)
-            if (!property_exists($plane, $prop))
-                $plane->$prop = $val;
```
Comment (achlin) [inline application/models/FleetModel.php]: Same as above.

File: application/models/FleetModel.php [hunk 3]
```diff
@@ -45,12 +52,33 @@ public function all()
...
+    * A fleet will be considered invalid and returns false if:
+    *   - The sum of the cost the planes in the fleet is overbudget (10 million)
+    */
+    public function validBudget($plane) {
+        $fleet = $this->all();
+        $fleetPrice = 0;
+
+        foreach($fleet as $kitePlane)
+        {
+            $wackyPlane = $this->wackyModel->getAirplane($kitePlane[key]);
```
Comment (achlin) [inline application/models/FleetModel.php]: I'm not sure we need another call to get the plane info from the wacky server. `$this->all()` should have given back the price already.

**Review by achlin — CHANGES_REQUESTED** [review]

File: application/models/FleetModel.php (hunk 1 above)
Comment (kculala) [reply application/models/FleetModel.php]: i wasn't sure what the previous code was doing so i tweaked it while testing something and forgot to undo. undid change.

File: application/models/FleetModel.php (hunk 2 above)
Comment (kculala) [reply application/models/FleetModel.php]: done

File: application/models/FleetModel.php (hunk 3 above)
Comment (kculala) [reply application/models/FleetModel.php]: fixed

File: application/models/FleetModel.php [hunk 4]
```diff
@@ -19,11 +19,14 @@ public function all()
...
         $planeIds = array_column($planes, 'airplaneCode');
-        $wackyPlanes = json_decode(file_get_contents(WACKY_SERVER_URI_BASE . '/airplanes'));
+        $wackyPlanes = $this->wackyModel->getAirplanes();
+
         $filteredPlanes = array_filter($wackyPlanes, function($p) use ($planeIds) {
             return in_array($p->id, $planeIds);
         });
 
+        $fleet = array();
+
```
Comment (achlin) [inline application/models/FleetModel.php]: Remove. Not used.

File: application/models/FleetModel.php [hunk 5]
```diff
@@ -36,6 +39,7 @@ public function all()
                 }
             }
         }
+
```
Comment (achlin) [inline application/models/FleetModel.php]: Remove

**Review by achlin — CHANGES_REQUESTED** [review]

**Review by achlin — APPROVED** [review]
